### ISSUE TITLE
Bad sound quality in HP chromebook 11A G8 EE and fedora 38

### ISSUE THREAD

**Issue by graywolf2** [WeirdTreeThing/chromebook-linux-audio#67]
To get work audio I downloaded the kernel 5.10.201 from kernel.org.
I got the config from kernel 5.10.23 in fedora 33.

I had set the following parameters in the kernel config.
CONFIG_DRM=y
CONFIG_DRM_AMDGPU=y
CONFIG_DEBUG_INFO=n

I commented all CONFIG_DEBUG_INFO_* parameters.

I also had set the CONFIG_INIT_STACK_ALL_ZERO=y
instead of CONFIG_INIT_STACK_NONE=y
because I hope this increase the kernel stability.
The kernel without CONFIG_INIT_STACK_ALL_ZERO=y spoiled important bios parameters after some time of work and fedora boot was blocked.
Btw the CONFIG_INIT_STACK_ALL_ZERO=y is used in kernel 6 from fedora.

I included the amdgpu/stoney_*.bin firmware in kernel.
I build and installed the kernel. I applied your script.

The sound quality in internal speakers is bad. For example the girl voice sounds like a old grandfather. The girl voice becomes very low frequency.

There are many controls in alsamixer. I tried to change every control but this do not help.

I tried the newer kernels than 5.10 but there is no internal sound card.

The bios firmware is official from google. I can not turn the bios wp off and use uefi firmware.
The seabios is used for boot fedora.

Grunt.
[debug-logs.tar.gz](https://github.com/WeirdTreeThing/chromebook-linux-audio/files/13684566/debug-logs.tar.gz)

**Comment by WeirdTreeThing** [comment]
There is likely some weird bug in these older kernels. I was able to compile and use 6.4 just fine on my careena with working sound.

**Comment by graywolf2** [comment]
In the past I checked some version 6 kernels including 6.1.62. No success. There was dummy output only in sound settings. There was HDMI sound device only in aplay -l output.
I will build and check the kernel 6.4.

**Comment by graywolf2** [comment]
I build and installed the kernel 6.4.12. I applied your script.
I got the same problems as I said in previous message.

Audio settings.
![Screenshot from 2023-12-23 18-22-56](https://github.com/WeirdTreeThing/chromebook-linux-audio/assets/12239412/24b9525f-8378-400d-94e9-6bce7a54d352)

Logs.
[debug-logs.tar.gz](https://github.com/WeirdTreeThing/chromebook-linux-audio/files/13759885/debug-logs.tar.gz)

Below is the output of some commands.

simple-user@fedora:~$ uname -r
6.4.16

simple-user@fedora:~$ aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: HDMI [HDA ATI HDMI], device 3: HDMI 0 [HDMI 0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: HDMI [HDA ATI HDMI], device 7: HDMI 1 [HDMI 1]
  Subdevices: 1/1
  Subdevice #0: subdevice #0

simple-user@fedora:~$ cat /proc/asound/cards
 0 [HDMI           ]: HDA-Intel - HDA ATI HDMI
                      HDA ATI HDMI at 0xf4c80000 irq 42

simple-user@fedora:~$ dmesg --level err
[    1.247186] kfd kfd: amdgpu: STONEY  not supported in kfd
[    2.932157] systemd[1]: bpf-lsm: Failed to load BPF object: No such process
[   11.033001] elants_i2c i2c-ELAN0001:00: nothing at this address
[   11.142631] raydium_ts i2c-RAYD0001:00: nothing at this address
[   11.982472] acp3x-alc5682-max98357 AMDI5682:00: DMIC gpio failed err=-2

**Comment by WeirdTreeThing** [comment]
Can you send me your kernel config

**Comment by graywolf2** [comment]
I got the config from kernel 6.4.12 in fedora repository. It is the file config-6.4.12-200.fc38.x86_64.

The kernel-6.4.12-config file is my modification of the kernel 6.4.12 original config.

I created the script for kernel build.
[configs-and-build-script.tar.gz](https://github.com/WeirdTreeThing/chromebook-linux-audio/files/13760199/configs-and-build-script.tar.gz)

The amdgpu/stoney_*.bin firmware is in ~/sources/firmware-from-fedora36 directory. The fedora 38 has the same amdgpu/stoney_*.bin firmware as fedora 36.
